api/v1beta1/dataplane_types.go

```diff
@@ -64,6 +65,75 @@ type DataPlaneOptions struct {
 
 	// +optional
 	Network DataPlaneNetworkOptions `json:"network"`
+
+	// +optional
+	Resources DataPlaneResources `json:"resources"`
+}
+
+// DataPlaneResources defines the resources that will be created and managed
+// for the DataPlane.
+type DataPlaneResources struct {
+	// PodDisruptionBudget is the configuration for the PodDisruptionBudget
+	// that will be created for the DataPlane.
+	PodDisruptionBudget *PodDisruptionBudget `json:"podDisruptionBudget,omitempty"`
+}
+
+// PodDisruptionBudget defines the configuration for the PodDisruptionBudget.
+type PodDisruptionBudget struct {
+	// Enabled indicates whether the PodDisruptionBudget should be created.
+	//
+	// +optional
+	// +kubebuilder:default=false
+	Enabled bool `json:"enabled"`
```

Comment:
@pmalek Why do we need this flag? As `resources.podDisruptionBudget` is optional, cannot we support enabling it simply by making it non-nil?

Reply:
My reasoning for this was to allow users to keep the spec but disable the pdb (for testing purposes or to temporarily suspend).

That use case might be contrived though.

Reply:
On a side note without the `enabled` field how do we consider the pdb enabled? If any of the spec fields is non nil? Or do we change the `spec` to be a pointer and check whether that's non nil?

Reply:
> My reasoning for this was to allow users to keep the spec but disable the pdb (for testing purposes or to temporarily suspend).

I'd drop it for simplicity and add only if there's a specific request for that. 

> On a side note without the enabled field how do we consider the pdb enabled? If any of the spec fields is non nil? Or do we change the spec to be a pointer and check whether that's non nil?

`DataPlaneResources.PodDisruptionBudget` is a pointer that can be made explicitly optional. If it is not `nil`, that means we should create a PDB based on the spec. Do I miss something?

### PDB disabled (`resources.podDistruptionBudget` pointer is nil)
```yaml
apiVersion: gateway-operator.konghq.com/v1beta1
kind: DataPlane
metadata:
  name: dataplane-with-pdb
spec:
  # Do not fill resources.podDisruptionBudget to not create a PDB for DP.
  # resources:
  #   podDisruptionBudget:
  deployment:
    replicas: 3
```

### PDB enabled with an empty spec (apparently it is allowed, I'm not sure what it effectively means if there's no minAvailable/maxUnavailable set)
```yaml
apiVersion: gateway-operator.konghq.com/v1beta1
kind: DataPlane
metadata:
  name: dataplane-with-pdb
spec:
  resources:
    podDisruptionBudget: {}
  deployment:
    replicas: 3
```

Reply:
👍 

Removed `enabled` field and added CEL validation as per the docs: https://kubernetes.io/docs/tasks/run-application/configure-pdb/#specifying-a-poddisruptionbudget

> You can specify only one of maxUnavailable and minAvailable in a single PodDisruptionBudget.